consensus/src/extension/mod.rs

```diff
@@ -1,189 +1,52 @@
-use futures::{FutureExt, StreamExt};
-
-use log::{debug, warn};
-
-use crate::{Hasher, NodeIndex, NodeMap, Receiver, Round, Sender, Terminator};
+use crate::{
+    dag::DagUnit,
+    units::{Unit, WrappedUnit},
+    Data, FinalizationHandler, Hasher, MultiKeychain,
+};
 
 mod election;
 mod extender;
 mod units;
 
 use extender::Extender;
 
-const LOG_TARGET: &str = "AlephBFT-extender";
-
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct ExtenderUnit<H: Hasher> {
-    creator: NodeIndex,
-    round: Round,
-    parents: NodeMap<H::Hash>,
-    hash: H::Hash,
-}
-
-impl<H: Hasher> ExtenderUnit<H> {
-    pub fn new(creator: NodeIndex, round: Round, hash: H::Hash, parents: NodeMap<H::Hash>) -> Self {
-        ExtenderUnit {
-            creator,
-            round,
-            hash,
-            parents,
-        }
-    }
-}
-
-/// A process responsible for executing the Consensus protocol on a local copy of the Dag.
-/// It receives units via a channel `electors` which are guaranteed to be eventually in the Dags
+/// A struct responsible for executing the Consensus protocol on a local copy of the Dag.
+/// It receives units which are guaranteed to be eventually in the Dags
```

Comment:
```suggestion
/// It receives units which are guaranteed to eventually appear in the Dag.
```
(resolvable)

Reply:
Fixed, although `Dags`, because it refers to all the nodes.